sys/can/pkt.c

```diff
@@ -0,0 +1,287 @@
+/*
```

Comment:
Maybe you can recap your thought about this problem so far at the next virtual developer meeting. What do you think? I think it's worth a more general discussion since other developers may need a similar data structure.